### DISCUSSION TITLE
OpenAPI examples incorrect for a Discriminated union?

### DISCUSSION THREAD

**Discussion by blueo** [fastapi/fastapi#11656] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, APIRouter
from pydantic import BaseModel, Discriminator, Tag
from typing import List, Union, Annotated, Any

app = FastAPI()

router = APIRouter(
    prefix="/api/v1/test",
    responses={404: {"description": "Not found"}},
)

def filter_discriminator(v: Any) -> str | None:
    if isinstance(v, dict):
        if v.get("date", None):
            return "date_filter"
        if v.get("tag", None):
            return "tag_filter"
    return None

class DateFilter(BaseModel):
    date: str

class TagFilter(BaseModel):
    tag: str

Filters = List[
        Annotated[
            Union[
                Annotated[DateFilter, Tag("date_filter")],
                Annotated[TagFilter, Tag("tag_filter")]
            ],
            Discriminator(filter_discriminator),
        ]
    ]

@router.post("/", status_code=200)
async def post_documents(
        filters: Filters
) -> str:
    return "test"
    
app.include_router(router)
```

### Description

When using a Discriminated Union the generated OpenAPI examples do not seem correct - it duplicates part of the first Union into the second.

Using the example above I get the following OpenAPI example generated:
![image](https://github.com/tiangolo/fastapi/assets/948122/07585f1e-1f34-44d5-8317-bae1ae664176)

it seems to persist for more union items - eg if I add a third item "test" then I'll get:
```JSON
[
  {
    "date": "string"
  },
  {
    "date": "string",
    "tag": "string"
  },
  {
    "date": "string",
    "test": "string"
  }
]
```

Am I doing something wrong with the Union definition perhaps? I've attempted to see if its a pydantic issue but it looks as though the schema is correct (doesn't duplicate the properties):
![image](https://github.com/tiangolo/fastapi/assets/948122/a5b22957-46e2-4942-9294-0f2a3197b7a6)

I'm not familiar with where this is generated in FastAPI but if someone could point me there I could try make a test case 

### Operating System

macOS

### Operating System Details

M1

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

3.11.9

**Comment by blueo** [comment]
As an Update, I tried throwing the openapi.json into the swagger-ui npm module directly and it generates correctly there:
![image](https://github.com/tiangolo/fastapi/assets/948122/2224c269-f792-49ea-acb4-d8d0519592f4)

reference swagger UI Code;
```Javascript
const express = require('express')
const pathToSwaggerUi = require('swagger-ui-dist').absolutePath()

const app = express()

app.use(express.static(pathToSwaggerUi))

app.listen(3000)
```

**Reply by blueo** [reply]
OK I tried the version of swagger ui that fastapi is running (5.9.0) and the issue is there - so looks like this is fixed in later swagger-ui versions. 
![image](https://github.com/tiangolo/fastapi/assets/948122/6131af08-0f55-4c66-9b80-197682176d4e)

looks like there is a way to get a newer version but its quite tricky: https://github.com/tiangolo/fastapi/discussions/10524

**Reply by gasscoelho** [reply]
Maybe this [discussion (#9287)](https://github.com/fastapi/fastapi/discussions/9287#discussioncomment-5905046) can give some light...

I had a similar issue and was able to work around it by using `Body`. So based on your example, it would be something like this:
```py
from fastapi import Body

Filters = List[
        Annotated[
            Union[
                Annotated[DateFilter, Tag("date_filter")],
                Annotated[TagFilter, Tag("tag_filter")]
            ],
            Body(discriminator=Discriminator(filter_discriminator)),
        ]
    ]
```

**Reply by blueo** [reply]
> ```python
> Body(discriminator=Discriminator(filter_discriminator)),
> ```

Hey thanks for the suggestion @gasscoelho but sadly it doesn't help - seems to be more about the Swagger UI output rather than a typing issue (given its working in later versions of the swagger ui)

**Reply by blueo** [reply, marked as answer]
actually looks like this may now be fixed as the version constraint on swagger UI is gone https://github.com/fastapi/fastapi/pull/11459
